### PR TITLE
Add tests for CorrectionWithGrad + jax.vmap

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

**Comment by eguiraud** [comment]
Current dependencies on/for this PR:
* main
  * **PR #44** <a href="https://app.graphite.dev/github/pr/eguiraud/correctionlib-gradients/44" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/eguiraud/correctionlib-gradients/44?utm_source=stack-comment).
